keps/sig-scheduling/5234-dra-resourceslice-mixins/README.md

```diff
@@ -0,0 +1,896 @@
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+None
+
+##### Unit tests
+
+<!--
+Generated with:
+
+go test -cover ./pkg/apis/resource/validation  ./staging/src/k8s.io/dynamic-resource-allocation/structured | sed -e 's/.*\(k8s.io[a-z/-]*\).*coverage: \(.*\) of statements/- `\1`: \2/' | sort
+
+-->
+
+- `k8s.io/dynamic-resource-allocation/structured`: `04/11/2025` - 91.3%
+- `k8s.io/kubernetes/pkg/apis/resource/validation`: `04/11/2025` - 97.8%
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+The integration test that verifies the theoretical maximum size of the ResourceSlice
+resource will be updated.
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+E2e tests will be added to verify that the mixins are properly flattened and
+used by the scheduler.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- 3 examples of real-world usage
+- Allowing time for feedback
+- [Conformance tests]
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+
+### Upgrade / Downgrade Strategy
+
+Mixins will no longer work when downgrading to a release without support for it.
+As described in the [Implementation](#implementation) section, devices that uses
+mixins or reference counter sets that is using mixins, will not be visible to the
+scheduler.
+
+### Version Skew Strategy
+
+During version skew where the apiserver supports the feature and the scheduler
+doesn't, the devices that is using the mixins feature will be dropped and
+not visible to the scheduler (ref [Implementation](#implementation)).
```

Comment:
This depends: with apiserver 1.34 + feature enabled and scheduler 1.33 (feature unknown), the apiserver sends devices with new fields, the scheduler doesn't know about the fields and processes incomplete devices.

This is only a problem for the initial alpha. For apiserver 1.35 + scheduler 1.34, the schedulers knows the fields and can do what is discussed under "Implementation".

So I think we are fine, we just need to document it a bit better.

Reply:
Yeah, that makes sense. I've updated the section to cover this scenario.